src/csharp/Microsoft.Spark/Utils/PythonSerDe.cs

```diff
@@ -67,5 +68,57 @@ internal static object[] GetUnpickledObjects(Stream stream, int messageLength)
                 ArrayPool<byte>.Shared.Return(buffer);
             }
         }
+
+        /// <summary>
+        /// Custom pickler for GenericRow objects.
+        /// Refer to
+        /// spark/sql/core/src/main/scala/org/apache/spark/sql/execution/python/EvaluatePython.scala
+        /// </summary>
+        internal class GenericRowPickler : IObjectPickler
+        {
+            private readonly string _module = "pyspark.sql.types";
+
+            public void Register()
+            {
+                Pickler.registerCustomPickler(GetType(), this);
+                Pickler.registerCustomPickler(typeof(GenericRow), this);
+            }
+
+            public void pickle(object o, Stream stream, Pickler currentPickler)
+            {
+                if (o.Equals(this))
+                {
+                    SerDe.Write(stream, Opcodes.GLOBAL);
+                    SerDe.Write(stream, Encoding.UTF8.GetBytes(
+                        $"{_module}\n_create_row_inbound_converter\n"));
+                }
+                else
+                {
+                    if (!(o is GenericRow genericRow))
+                    {
+                        throw new InvalidOperationException("A GenericRow object is expected.");
+                    }
+
+                    currentPickler.save(this);
+                    SerDe.Write(stream, Opcodes.TUPLE1);
+                    SerDe.Write(stream, Opcodes.REDUCE);
+
+                    SerDe.Write(stream, Opcodes.MARK);
+                    for (int i = 0; i < genericRow.Size(); ++i)
+                    {
+                        currentPickler.save(genericRow.Get(i));
+                    }
+
+                    SerDe.Write(stream, Opcodes.TUPLE);
+                    SerDe.Write(stream, Opcodes.REDUCE);
+                }
+            }
+        }
+
+        internal static Pickler CreatePickler()
+        {
+            new GenericRowPickler().Register();
+            return new Pickler();
```

Comment:
Do we need to add unpickling on C# side for the returned Row object?

Reply:
Removed. Thanks!